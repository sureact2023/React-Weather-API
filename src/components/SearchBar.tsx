import React, { ChangeEventHandler, Dispatch, FC, SetStateAction, useState } from "react";
import { TextField } from "@mui/material";

interface iSearchBar {
  setCityName: Dispatch<SetStateAction<string>>;
}

const SearchBar: FC<iSearchBar> = ({ setCityName }) => {
  const [search, setSearch] = useState("");
  return (
    <form
      onSubmit={(e) => {
        e.preventDefault();
        setCityName(search);
      }}
    >
      <TextField
        fullWidth
        id="outlined-basic"
        label="Search city name here"
        variant="outlined"
        onChange={(e)=>setSearch(e.target.value)}
      />
    </form>
  );
};

export default SearchBar;
