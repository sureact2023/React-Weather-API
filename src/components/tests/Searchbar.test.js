import { render, screen } from "@testing-library/react";
import SearchBar from "../SearchBar";


describe("Testing the Search Component", () => {
    const setCityName = jest.fn()

    it('Snapshot the component', () => {
        const { container } = render(<SearchBar setCityName={setCityName}/>)
        expect(container).toMatchSnapshot()
    })

    it('expect the function to be called',()=>{
        setCityName.call()
        expect(setCityName).toBeCalled()
    })

});