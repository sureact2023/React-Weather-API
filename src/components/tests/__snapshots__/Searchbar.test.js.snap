// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Search Component Snapshot the component 1`] = `
<div>
  <form>
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="outlined-basic"
        id="outlined-basic-label"
      >
        Search city name here
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="outlined-basic"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-yjsfm1"
          >
            <span>
              Search city name here
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </form>
</div>
`;
